### PR TITLE
Advance InkProof B006, B007

### PR DESCRIPTION
**Pull request by JBenda**
Try to accomplish as much InkProofnes with the current feature set as possible.

### CODE REVIEW

File: inkcpp/value.cpp [hunk 1]
```diff
@@ -303,6 +303,82 @@ namespace ink
...
+					break;
+				case data_type::string_table_pointer:
+				case data_type::allocated_string_pointer:
+					c = d.string_val;
+					break;
+				default: number[0] = 0;
+				}
+			}
+			bool value::compare_string(const value& left, const value& right) {
+				size_t l_i = 0;
```
Comment (brwarner) [inline inkcpp/value.cpp:329]: This function looks great! Thanks for solving this long untouched problem of string comparison. If you could add a comment above each if/else/while that'd be great for readability.

**Review by brwarner — APPROVED** [review]
Looks great. Clever string compare. Might be nice to have a few unit tests for it in the future but I'm happy with everything for now.

**Comment by JBenda** [comment]
Fair point, but isn't this with ink-proof kind of redundant?
I will implement some test ^^ (soon TM)

**Comment by brwarner** [comment]
I guess I'm not sure how extensively ink-proof would test problem cases in
this algorithm.

Something like comparing (& is string append)

4 & 55 & "hello"
with
45 & "5hello"

Is the kind of case you'd want to write a test for knowing how your
algorithm works. It's not the kind of test ink-proof is going to have.

**Pull request merged**